cmd/object-api-utils.go

```diff
@@ -239,10 +239,35 @@ func getCompleteMultipartMD5(parts []CompletePart) string {
 func cleanMetadata(metadata map[string]string) map[string]string {
 	// Remove STANDARD StorageClass
 	metadata = removeStandardStorageClass(metadata)
+	// Remove Tags
+	metadata = removeTags(metadata)
 	// Clean meta etag keys 'md5Sum', 'etag', "expires".
 	return cleanMetadataKeys(metadata, "md5Sum", "etag", "expires")
 }
 
+// removeTags deletes all tag entries (that start with x-amz-tag-)
+// from the given map and returns the same map
+func removeTags(metadata map[string]string) map[string]string {
```

Comment:
Always make a copy of map to avoid races

Reply:
done